teaching.rb

```diff
@@ -0,0 +1,19 @@
+module Teaching
+
+  def teach_stuff
+  	if self.class::TEACHER_TYPE == "Apprentice"
```

Comment:
Nope nope nope. Big red flag.  Never check what kind of thing you are from a module or superclass.  If each subclass/thing that includes the module needs different behavior, just leave the behavior in the respective classes.  Look up duck typing.